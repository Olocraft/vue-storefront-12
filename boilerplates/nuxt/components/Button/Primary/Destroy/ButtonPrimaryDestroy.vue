<template>
  <button
    v-focus
    class="inline-flex items-center justify-center cursor-pointer bg-rose-600 hover:bg-rose-700 active:bg-rose-800 disabled:bg-gray-200 text-white disabled:text-gray-500 disabled:opacity-50 uppercase rounded-md font-body text-base font-medium px-[16px] py-2 shadow-base hover:shadow-medium disabled:shadow-none disabled:cursor-not-allowed outline-violet"
    v-on="$listeners"
  >
    <slot />
  </button>
</template>

<script>
import { focus } from '../../../../utils/focus-directive.js';

export default {
  name: 'ButtonPrimaryDestroy',
  directives: {
    focus
  },
  props: {
    disabled: {
      type: Boolean,
      default: false
    }
  }
};
</script>
