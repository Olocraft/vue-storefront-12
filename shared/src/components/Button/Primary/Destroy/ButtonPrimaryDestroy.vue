<template>
  <button
    v-focus
    class="inline-flex items-center justify-center cursor-pointer bg-rose-600 hover:bg-rose-700 active:bg-rose-800 disabled:bg-gray-200 text-white disabled:text-gray-500 disabled:opacity-50 uppercase rounded-md font-body text-base font-medium px-[16px] py-2 shadow hover:shadow-md disabled:shadow-none disabled:cursor-not-allowed outline-violet"
  >
    <slot />
  </button>
</template>

<script>
import { focus } from '../../../../utils/focus-directive.js';

export default {
  name: 'ButtonPrimaryDestroy',
  directives: {
    focus
  },
  props: {
    disabled: {
      type: Boolean,
      default: false
    }
  }
};
</script>
