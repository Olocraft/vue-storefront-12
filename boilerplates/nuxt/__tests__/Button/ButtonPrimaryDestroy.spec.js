import { shallowMount } from '@vue/test-utils';
import ButtonPrimaryDestroy from '../../components/Button/Primary/Destroy/ButtonPrimaryDestroy.vue';

let wrapper;

describe('ButtonPrimaryDestroy.vue', () => {
  beforeEach(() => {
    wrapper = shallowMount(ButtonPrimaryDestroy);
  });
  it('render component correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
