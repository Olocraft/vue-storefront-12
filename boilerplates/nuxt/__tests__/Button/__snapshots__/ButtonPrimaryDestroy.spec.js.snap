// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPrimaryDestroy.vue render component correctly 1`] = `<button class="inline-flex items-center justify-center cursor-pointer bg-rose-600 hover:bg-rose-700 active:bg-rose-800 disabled:bg-gray-200 text-white disabled:text-gray-500 disabled:opacity-50 uppercase rounded-md font-body text-base font-medium px-[16px] py-2 shadow-base hover:shadow-medium disabled:shadow-none disabled:cursor-not-allowed outline-violet"></button>`;
